internal/server/warcserver/db.go

```diff
@@ -299,6 +227,95 @@ func (c DbCdxServer) sortedParallelSearch(api *CdxjServerApi, perCdxFunc PerCdxF
 	return count, err
 }
 
+func (c DbCdxServer) unsortedSerialSearch(api *CdxjServerApi, perCdxFunc PerCdxFunc) (int, error) {
+	keys := make([]string, len(api.Urls))
+	for i, url := range api.Urls {
+		key := parseKey(surt.UrlToSsurt(url), api.MatchType)
+		keys[i] = key
+	}
+
+	count := 0
+	err := c.Walk(func(txn *badger.Txn) error {
+		// initalize badger iterators
+		iterators := make([]*badger.Iterator, len(keys))
+		prefixes := make([][]byte, len(keys))
+		for i, key := range keys {
+			prefixes[i] = []byte(key)
+			opts := badger.DefaultIteratorOptions
+			opts.Prefix = prefixes[i]
+
+			iterators[i] = txn.NewIterator(opts)
+			defer iterators[i].Close()
+
+			iterators[i].Seek(prefixes[i])
+		}
+
+		for len(iterators) > 0 {
+			// set timestamp to aprox max time.Time value
+			earliestTimestamp := time.Unix(1<<62, 1<<62)
+			earliestIndex := -1
+			// find the earliest timestamp
+			for i, iter := range iterators {
+				// if iter is no longer valid, close it, remove it from slice and restart search
+				if !iter.ValidForPrefix(prefixes[i]) {
+					iteratorsLen := len(iterators)
+					iterators[i].Close()
+					// remove iterator from list
+					iterators[i] = iterators[iteratorsLen-1]
+					iterators = iterators[0 : iteratorsLen-1]
+					break
+				}
+
+				item := iter.Item()
+				// in the event of parse error we get a zero timestamp
+				timestamp, err := time.Parse(timeLayout, Key(item.Key()).ts())
```

Comment:
Name of variable `timestamp` collides with name of imported package. I suggest naming it `ts`.